src/WikiService.Validation/Rubric/EditRubricRequestValidator.cs

```diff
@@ -0,0 +1,154 @@
+      AddСorrectOperations(nameof(EditRubricRequest.IsActive), new () { OperationType.Replace });
+
+      #endregion
+
+      #region Name
+
+      AddFailureForPropertyIf(
+        nameof(EditRubricRequest.Name),
+        x => x == OperationType.Replace,
+        new()
+        {
+          {
+            x => !string.IsNullOrWhiteSpace(x.value?.ToString()),
+            "Name must not be empty."
+          },
+          {
+            x => x.value.ToString().Trim().Length <= 150,
+            "Name is too long."
+          }
+        }, CascadeMode.Stop);
+
+      #endregion
+
+      #region IsActive
+
+      AddFailureForPropertyIf(
+       nameof(EditRubricRequest.IsActive),
+       x => x == OperationType.Replace,
+       new()
+       {
+         {
+           x => bool.TryParse(x.value?.ToString(), out bool _), "Incorrect rubric IsActive format"
+         },
+       }, CascadeMode.Stop);
+
+      #endregion
+
+      #region ParentId
+
+      await AddFailureForPropertyIfAsync(
+        nameof(EditRubricRequest.ParentId),
+        x => x == OperationType.Replace,
+        new()
+        {
+          {
+            async (x) =>
+            {
+              if (x.value?.ToString() is null)
+              {
+                return true;
+              }
+
+              if (!Guid.TryParse(x.value?.ToString(), out Guid parentId))
+              {
+                return false;
+              }
+
+              return await _rubricRepository.DoesExistAsync(parentId);
+            },
+            "Parent id doesn`t exist."
+          }
+        }, CascadeMode.Stop);
+    }
+
+    #endregion
+
+    public EditRubricRequestValidator(
+        IRubricRepository rubricRepository)
+    {
+      _rubricRepository = rubricRepository;
+
+      RuleForEach(x => x.Item2.Operations)
+        .CustomAsync(async (x, context, _) => await HandleInternalPropertyValidationAsync(x, context));
+
+      When(x => x.Item2.Operations.Any(o =>
+        (o.path.EndsWith(nameof(EditRubricRequest.ParentId), StringComparison.OrdinalIgnoreCase))
+        || (o.path.EndsWith(nameof(EditRubricRequest.Name), StringComparison.OrdinalIgnoreCase))),
+        () =>
+        {
+          RuleFor(x => x)
+           .MustAsync(async (x, _) =>
+           {
+             Guid? _currentParentId = x.Item1.ParentId;
+             string _currentRubricName = x.Item1.Name;
+
+             foreach (Operation<EditRubricRequest> item in x.Item2.Operations)
+             {
+               if (item.path.EndsWith(nameof(EditRubricRequest.ParentId), StringComparison.OrdinalIgnoreCase))
+               {
+                 if (Guid.TryParse(item.value?.ToString(), out Guid parentId) || item.value is null)
+                 {
+                   _currentParentId = item.value is null
+                    ? null
+                    : parentId;
+                 }
+                 else
+                 {
+                   return false;
```

Comment:
what would happen if ParentId has bad format? would user receive error message "That name already exists."?
May be "return false" or remove "else" at all?